.github/workflows/tests.yml

```diff
@@ -9,19 +9,28 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6', '3.8', '3.10', '3.12']
+
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.8
+
+    - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v2
       with:
-        python-version: 3.8
-    - name: Install PyYAML
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
       run: |
-         pip3 install -r requirements.txt
-         pip3 install coverage
+         pip install -r requirements.txt
+         pip install coverage
+
     - name: Test error outputs
```

Comment:
@Myrausman we should keep `pip3` as it will intend to `python 3.x` packages and it is specifically associated with `Python 3.x` versions. 
While the only `pip` refers to the default package manager for `Python 2.x`. Additionally, on systems where `Python 2` is installed, `pip` installs packages for `Python 2`. So, in order to avoid conflicts between `Python 2.x` and `Python 3.x`, we should keep as `pip3`.